### ISSUE TITLE
Update to translation table, addition of 'Teacher Tips' column

### ISSUE THREAD

**Issue by oliver-phet** [phetsims/website#324]
@mattpen Here is the mockup of the new translation table. 
Note the changes to the 'Language' column of the table: the new 'All [locale]' replacing 'All sims in [locale]'

<img width="904" alt="screen shot 2015-11-19 at 12 12 07 pm" src="https://cloud.githubusercontent.com/assets/5090275/11281210/dcd6570a-8eb6-11e5-910d-393cac1865ed.png">

**Comment by amanda-phet** [comment]
Just so the "All..." links are clear, I made an updated mockup. 

Also, a teacher tips PDF icon will only be there if applicable for that locale, and the icon should be the same image used for teacher tips under For Teachers. 

<img width="694" alt="screen shot 2015-11-23 at 1 49 55 pm" src="https://cloud.githubusercontent.com/assets/7852916/11349394/475cb972-91e9-11e5-830b-030566a83f2f.png">

**Comment by mattpen** [comment]
@oliver-phet What is your current process for uploading the teacher tips?  As of today, the teacher tips do not have a locale associated.  I can update the database so it has a locale column, but I want to make sure that when they are uploaded that we can specify the locale.

**Comment by oliver-phet** [comment]
@mattpen The sim designer uploads the tips, I've never actually done it before. Perhaps @arouinfar or @amanda-phet can comment. I think it would be fine to add a 'locale' column that is required - so the original tips are uploaded as "_en".

**Comment by amanda-phet** [comment]
@oliver-phet that's correct. We upload them under administration. I agree that a locale selection is a good idea since they go to different places.

**Comment by arouinfar** [comment]
@oliver-phet I also think it'd be good to add a locale section.

**Comment by mattpen** [comment]
Are we keeping the Teacher Tips section as it currently exists on the main site (in the "For Teachers" section)?  

If so, if an end user views the simulation page in a locale that is not English and a teacher tips file exists for that sim in that locale, should this link go to the English guide or the guide for the locale?

**Comment by oliver-phet** [comment]
Existing Teacher Tips section will not be changing... except if an end user views the simulation page in a locale that is not English and a teacher tips file exists for that sim in that locale, it will open the guide in that locale.

**Comment by mattpen** [comment]
I think I've got this working, but I can't test the file uploads with my VM due to a separate issue.  I'll deploy to phet-dev once help-center testing is complete.

**Comment by mattpen** [comment]
@oliver-phet - I've adjusted the translations table and added the option to upload translations by locale.  There is a new section in the Simulations admin page.  This is now up on phet-dev if you want to take a look or test it out.

**Comment by mattpen** [comment]
I added a couple of empty pdf's to alpha decay to test it out.

**Comment by amanda-phet** [comment]
I'm finding these tables to be a bit problematic. Different sims seem to show different column widths, and in this extreme case the "Download or Run" title doesn't line up at all with the column. I also worry that the translated sims are not getting enough room when that is probably the most important content in this table. Is it because the first few columns can get too wide? can we put a max width on them?

<img width="619" alt="screen shot 2015-12-22 at 11 42 32 am" src="https://cloud.githubusercontent.com/assets/7852916/11963050/6ee20744-a8a1-11e5-8204-44206689d630.png">

**Comment by mattpen** [comment]
@amanda-phet We can put a max width on the first two columns, but then they will wrap around to a second line.  There's just not enough room in the main-content section of the page to format the table like Oliver posted.  Is the third column necessary?  Other than the word "All" it seems to be a duplicate of the first column.  I could eliminate the third column and make the second column link to the page for the locale.

In regards to the headers, I can reformat these today.

**Comment by amanda-phet** [comment]
> I could eliminate the third column and make the second column link to the page for the locale. 

That's a good idea. I will bring that up to the team tomorrow. Thanks!

**Comment by mattpen** [comment]
As there was some confusion, I'm clarifying which of the in-progress website changes are dependent on each other.  
This change is not dependent on any others.

**Comment by arouinfar** [comment]
@mattpen's mention of his work on this during status got me thinking about this issue.  Currently, we have two translated teacher tips living in the activities -- Under Pressure (Java) and Balancing Chemical Equations (HTML5). Under Pressure has a Translations table to move the tips to, but BCE does not since it hasn't been redeployed yet. I have a few questions...
* What happens if someone submits translated tips for non-translatable sims? 
* What happens if someone submits translated tips for a sim that does not exist in that locale, but has other translations? 
* Should we simply leave the Download or Run column empty in cases where translated tips exist, but not a translated sim?

Perhaps this is something we should discuss at the next website meeting @amanda-phet.

**Comment by amanda-phet** [comment]
@arouinfar good questions. Since we aren't meeting for 2 weeks, let's discuss over email in the meantime.

**Comment by amanda-phet** [comment]
The consensus for the edge cases that @arouinfar brought is to hold on to translated teacher tips for sims that have not yet been translated (waiting on redeployment, for example) and upload translated teacher tips to the translated sims table *even if the sim is not translated in that locale* (there won't be a link to download/run the sim in that locale, but the tips document will be available anyway).

**Comment by mattpen** [comment]
@amanda-phet @arouinfar I deployed the latest version of the table to phet-dev today.  I have not yet implemented the behavior to add a row for locale that do not have a translated sim but have a translated teacher's guide.  This will be somewhat difficult to implement, but I'll see what I can do.

**Comment by amanda-phet** [comment]
> I have not yet implemented the behavior to add a row for locale that do not have a translated sim but have a translated teacher's guide.

I don't think we need to wait on that functionality to deploy this change. What do you think @arouinfar ?

**Comment by arouinfar** [comment]
@amanda-phet I agree. Now that BCE has been redeployed, we don't currently have the problem of translated tips for not-yet-translatable sims.

**Comment by amanda-phet** [comment]
@arouinfar I think Matt is saying the sim would still need to be translated into that locale for the tips to be visible. Don't we have a translated tips document for a sim that is not translated into that locale? In that case, the tips doc would still not be in the table.

**Comment by arouinfar** [comment]
Got it @amanda-phet. The teacher tips for BCE has been translated into Spanish (possibly Colombian Spanish given the author's affiliation). There is no Spanish or Colombian Spanish translation of the sim, though there is a Peruvian Spanish translation. How picky should we be when it comes to sub-locales? 

I would think that unless there is a Peruvian Spanish version of the teacher tips, ANY Spanish version of the tips could populate the teacher tips in the row for Peruvian Spanish. I think it's unlikely that we'll end up with multiple teacher tips translations with the same base locale, especially if there is only a trusted translator for one sub-locale (e.g. teacher tips in es_CO, es_MX, and es_ES but a sim only available in es_PE). 

If we are regularly receiving translated teacher tips for a locale without a trusted translator, we could encourage the tips translator to start translating sims too.

**Comment by amanda-phet** [comment]
Sounds good to me! I couldn't remember exactly which tips we had, but given the current situation it sounds like we can just use our discretion.

**Comment by arouinfar** [comment]
I think the only translated tips we have are the Java version of Under Pressure and the HTML5 version of Balancing Chemical Equations. Both are in Spanish (possibly Colombian Spanish). Under Pressure has es and es_PE translations, and BCE has just a es_PE translation.

**Comment by mattpen** [comment]
@amanda-phet @arouinfar  Are there any instances where we received a guide for a locale where there was no sim translated into a related sub-locale?

You can select the locale when you upload the guide, would it be problematic to just upload it in the locale for the sim with the closest sub_locale? In the BCE example, you could just select es_PE as the locale instead of es_CO.

If we've only received two of them, this sounds like a better solution than trying to implement a locale fallback automation procedure (which would take time to develop, test, troubleshoot, etc).  Are we expecting to receive a lot of these?

**Comment by arouinfar** [comment]
@mattpen As far as I know, we don't have any translated tips for sims that haven't been translated into a related sub-locale. I don't think we need to worry about any automated procedures for now. We can just use our best judgement when manually uploading the teacher tips.

**Comment by mattpen** [comment]
@oliver-phet @arouinfar @amanda-phet This should be in a final state now on phet-dev. I opened a separate issue to test it out.

**Comment by mattpen** [comment]
@ariel-phet @amanda-phet - Letting the columns have dynamic width on this table is not working out so well -- [expand the Translation section here](https://phet-dev.colorado.edu/en/simulation/arithmetic).  The best solution IMO is to choose a fixed width in either percentage or pixels for the columns, although I'm open to suggestion.  Setting a minimum width is another option but is more difficult to implement and possibly introduces other layout bugs.  I think the widths for [acid-base-solutions](https://phet-dev.colorado.edu/en/simulation/acid-base-solutions) look pretty good, but I'd like to confirm with a designer.

Setting priority to 3 based on sype discussion with @ariel-phet.

**Comment by ariel-phet** [comment]
@mattpen as a first step can we change "All sims in" to "All" that will tight things up a bit, plus if someone translates that string it could get very long, ie french....(tout le simulations dans...), then lets take a closer look

**Comment by ariel-phet** [comment]
@mattpen also once you have changed to "all" I would try the fixed width like in acid base-solutions but a little tighter (less width) in the language section (since we will have saved some space) which will leave a little more room for the teacher tips column. 5 percent or so more room in the teacher tips column might be nice

**Comment by mattpen** [comment]
@ariel-phet - I published a first attempt to phet-dev.  Do the column widths look ok? I was thinking that the Translated Sim Title column could be moved a little further over to the right, but the headers are pretty tight as is in some locales, pt_BR for example:
![image](https://cloud.githubusercontent.com/assets/11747676/13308173/62d81244-db2e-11e5-8097-08b357b01b13.png)


The "Teacher Tips" header definitely needs to be fixed in RTL.  Otherwise do you see anything else to fix?

**Comment by ariel-phet** [comment]
@mattpen agreed the title column should be moved to the right, actually a fair bit it looks like (at least 15 percent or so?)

Is there a way to enforce a maximum heading size the way we do i18n in sims? So that if the heading goes beyond a certain width the font size shrinks down? That would likely be more aesthetically pleasing than titles overlapping.

**Comment by ariel-phet** [comment]
Perhaps we go with something like this? And simplify "teacher tips" in the table to just "tips"? In English it definitely helps with the alignment and such since the word Tips matches the size of the pdf icon so much better. 

![table](https://cloud.githubusercontent.com/assets/4051174/13308410/3a5f1bda-db30-11e5-8dcc-03b2a9e9913e.PNG)

**Comment by mattpen** [comment]
| Is there a way to enforce a maximum heading size the way we do i18n in sims? So that if the heading goes beyond a certain width the font size shrinks down? That would likely be more aesthetically pleasing than titles overlapping.

Agreed that long term this is probably a better solution, but this would take some time to implement as this style doesn't exist with an HTML/CSS solution.  It would require writing a script to detect and adjust large translated strings.  

I'll just try shortening the title to "TIPS" for now.

**Comment by mattpen** [comment]
@ariel-phet I widened the "All sims in" column and changed the String for teacher tips.  This is published on phet-dev.  Thoughts?

**Comment by ariel-phet** [comment]
Looks good to me in English.

Is there a way to "reset" the word to "tips" so in other languages we don't get this sort of thing happening?

![tips](https://cloud.githubusercontent.com/assets/4051174/13311175/6bd1ee40-db49-11e5-9790-1861aa417d3b.PNG)

**Comment by ariel-phet** [comment]
basically so translators are forced to restranslate that word?

**Comment by mattpen** [comment]
Yes, I can change the key for the string so that it doesn't have any translations.

**Comment by ariel-phet** [comment]
OK, lets change the key for "teacher tips" and also for "all the sim in" so they become just "tips" and "all"

**Comment by mattpen** [comment]
@ariel-phet - I reset the keys for the "All sims in" and "Teacher Tips" strings, so they should now be one word in all languages.  This change was pushed to phet-dev.

**Comment by ariel-phet** [comment]
@mattpen I think this change is good to go!

**Comment by ariel-phet** [comment]
Hmm, actually one thing (sorry...) I see in the mockups above the font was smaller and the letters all lowercases. I like them all uppercase, it seems aesthetically a good choice, but could we try it with the font down by 1 pt size to save a little room?

**Comment by mattpen** [comment]
@ariel-phet - The mockups in this issue are from before we completed the table standardization in #347.  The font is now controlled in the one place across all the tables on the website.  Should I (1) add an exception for this table or (2) change the font size on tables site wide?  If we choose (1) this table may ignore site-wide changes later or and if we choose (2) the change would require a little extra QA/pixel-polish.

**Comment by ariel-phet** [comment]
No, go ahead and deploy as is, we can revisit sometime if we are seeing a major problem, but everything appears to be OK in different locales and the current font size works well in terms of legibility and such.

**Comment by mattpen** [comment]
OK, thanks.  I'm planning to deploy tonight.

**Comment by mattpen** [comment]
This was deployed to figaro tonight.
